### PR TITLE
[SRVC-17749] Compressed all the changes around adding CompactIndexApi

### PR DESCRIPTION
**Pull request by brand-it**
#### Source
* https://github.com/geminabox/geminabox/pull/547

### CODE REVIEW

File: lib/geminabox/server.rb [hunk 1]
```diff
@@ -109,6 +72,21 @@ def file_class=(klass)
       query_gems.any? ? gem_list.to_json : {}
     end
 
+    get '/names' do
+      content_type 'text/plain'
+      with_etag_for(CompactIndexApi.new.names)
+    end
+
+    get '/versions' do
```
Comment (bwanicur-wm) [inline lib/geminabox/server.rb:80]: 🎉 👏 👏 👏 - woot!

**Review by jcraigk — APPROVED** [review]
🚢

**Review by commitshappen — APPROVED** [review]
yolo

**Comment by commitshappen** [comment]
if it doesn't work out, we can always go back to source 🤷🏼

**Comment by brand-it** [comment]
> if it doesn't work out, we can always go back to source 🤷🏼

true, however a solid week of testing this on my personal computer and after reading over all the code there is nothing I can find that I would not have done my self

**Review by jscanish — APPROVED** [review]
### QA Approved

Confirmed that geminabox server runs. Pointed local bundle to it and ran `bundle install` to confirm it still works.
Screenshot 2023-06-06 at 1 14 53 PM

**Review by alexcoll — APPROVED** [review]

**Pull request merged**